### PR TITLE
Migrated from FxCop analyzers to .NET analyzers.

### PR DESCRIPTION
**Pull request by MoFtZ**
https://docs.microsoft.com/en-us/visualstudio/code-quality/migrate-from-fxcop-analyzers-to-net-analyzers?view=vs-2019

### CODE REVIEW

File: Src/ILGPU/ILGPU.csproj [hunk 1]
```diff
@@ -50,54 +51,54 @@
             <DependentUpon>AtomicFunctions.tt</DependentUpon>
         </None>
         <None Include="Backends\PTX\PTXIntrinsics.Generated.cs">
-          <DesignTime>True</DesignTime>
-          <AutoGen>True</AutoGen>
-          <DependentUpon>PTXIntrinsics.Generated.tt</DependentUpon>
+            <DesignTime>True</DesignTime>
```
Comment (m4rs-mt) [inline Src/ILGPU/ILGPU.csproj:54]: Question: should we create a different PR (or alternatively a commit) in order to track the re-formatting operations separately?

**Review by m4rs-mt — APPROVED** [review]

File: Src/ILGPU/ILGPU.csproj (hunk 1 above)
Comment (MoFtZ) [reply Src/ILGPU/ILGPU.csproj:54]: These changes are unintended. They should be reverted.

**Review by m4rs-mt — CHANGES_REQUESTED** [review]
After reverting these unintentional changes, this should be fine. 🚀

**Comment by MoFtZ** [comment]
Unintended changes removed.

**Review by m4rs-mt — APPROVED** [review]

**Comment by MoFtZ** [comment]
Rebased to fixed merge conflicts.

**Pull request merged**